File: Sorting/Sorting in CPP/insertionSort.cpp
//PROGRAM FOR INSERTION SORT: DAA
#include<iostream> 
using namespace std; 
void insertionSort(int arr[], int n)  
{  
    int i, k, j;  
    for (i = 1; i < n; i++) 
    {  
        k = arr[i];  
        j = i - 1;  
        while (j >= 0 && arr[j] > k) 
        {  
            arr[j + 1] = arr[j];  
            j = j - 1;  
        }  
        arr[j + 1] = k;  
    }  
} 
void prntArr(int arr[], int n)  
{  
    int i;  
    for (i = 0; i < n; i++)  
    cout << arr[i] << " ";  
    cout << endl; 
}  

//MainFunction
int main()  
{  
    int n;
    cout<<"Enter the size of array: ";
    cin>>n;
    int arr[n];
    cout<<"Enter the array elements: ";
    for(int i=0;i<n;i++)
    {
        cin>>arr[i];
    } 
    insertionSort(arr, n);  
    prntArr(arr, n);  
  
    return 0;  
}  

File: Algorithms/Sorting/Sorting_Java/InsertionSort.java
import java.util.Scanner;

public class InsertionSort {
    public static void insertionSort (int[] a) {
        int temp, ind, n = a.length;
        // Loop through the array
        for (int i = 0; i < n; ++i) {
            ind = i;
            for (int j = i - 1; j >= 0; --j) {
                if (a[j] > a[ind]) {
                    temp = a[j];
                    a[j] = a[ind];
                    a[ind] = temp;
                    ind = j;
                }
            }
        }
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        int n;
        System.out.print("Enter the size of the array: ");
        n = in.nextInt();

        int[] arr = new int[n];
        System.out.println("Enter array elements:");
        for (int i = 0; i < n; ++i) {
            arr[i] = in.nextInt();
        }

        // Apply the insertion sort algo
        insertionSort(arr);

        System.out.println("The sorted array is: ");
        for (int i = 0; i < n; ++i) {
            System.out.print(arr[i] + " ");
        }
    }
}
